File: 백준/Bronze/23027. 1번부터 문제의 상태가…？/1번부터 문제의 상태가…？.c
#include <stdio.h>
#include <string.h>

int main() {
    char S[1001];
    fgets(S, sizeof(S), stdin);
    size_t len = strlen(S);
    if (S[len - 1] == '\n') {
        S[len - 1] = '\0';
        len--;
    }
    int hasA = 0, hasB = 0, hasC = 0;
    for (size_t i = 0; i < len; i++) {
        if (S[i] == 'A') {
            hasA = 1;
            break;
        } else if (S[i] == 'B') {
            hasB = 1;
        } else if (S[i] == 'C') {
            hasC = 1;
        }
    }
    if (hasA) {
        for (size_t i = 0; i < len; i++) {
            if (S[i] == 'B' || S[i] == 'C' || S[i] == 'D' || S[i] == 'F') {
                S[i] = 'A';
            }
        }
    } else if (hasB) {
        for (size_t i = 0; i < len; i++) {
            if (S[i] == 'C' || S[i] == 'D' || S[i] == 'F') {
                S[i] = 'B';
            }
        }
    } else if (hasC) {
        for (size_t i = 0; i < len; i++) {
            if (S[i] == 'D' || S[i] == 'F') {
                S[i] = 'C';
            }
        }
    } else {
        for (size_t i = 0; i < len; i++) {
            S[i] = 'A';
        }
    }
    printf("%s\n", S);
    return 0;
}

File: 백준/Bronze/23027. 1번부터 문제의 상태가…？/1번부터 문제의 상태가…？.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        char[] S = input.toCharArray();
        boolean hasA = false;
        boolean hasB = false;
        boolean hasC = false;
        for (char c : S) {
            if (c == 'A') {
                hasA = true;
                break;
            } else if (c == 'B') {
                hasB = true;
            } else if (c == 'C') {
                hasC = true;
            }
        }
        if (hasA) {
            for (int i = 0; i < S.length; i++) {
                if (S[i] == 'B' || S[i] == 'C' || S[i] == 'D' || S[i] == 'F') {
                    S[i] = 'A';
                }
            }
        } else if (hasB) {
            for (int i = 0; i < S.length; i++) {
                if (S[i] == 'C' || S[i] == 'D' || S[i] == 'F') {
                    S[i] = 'B';
                }
            }
        } else if (hasC) {
            for (int i = 0; i < S.length; i++) {
                if (S[i] == 'D' || S[i] == 'F') {
                    S[i] = 'C';
                }
            }
        } else {
            for (int i = 0; i < S.length; i++) {
                S[i] = 'A';
            }
        }
        System.out.println(new String(S));
    }
}